MdeModulePkg/Bus/Pci/NvmExpressDxe/UnitTest/MediaSanitizeUnitTest.c

```diff
@@ -0,0 +1,1208 @@
+/** @file -- MediaSanitizeUnitTest.c
+  Placeholder/framework for developing a Media Sanitize unit test package.
+
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <Uefi.h>
+#include <Library/BaseLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/UefiLib.h>
+#include <Library/DebugLib.h>
+#include <Library/UnitTestLib.h>
+#include <Library/UefiBootServicesTableLib.h>
+#include <Library/UefiApplicationEntryPoint.h>
+#include <Library/UefiRuntimeServicesTableLib.h>
+#include <Uefi/UefiBaseType.h>
+#include <Library/UefiBootManagerLib.h>
+#include <Protocol/NvmExpressPassthru.h>
+
+#include "../NvmExpress.h"
+#include "../NvmExpressBlockIo.h"
+#include "../NvmExpressMediaSanitize.h"
+#include "../NvmExpressHci.h"
+#include "../../../Include/Protocol/MediaSanitize.h"

```

Comment:
Agreed. Will update.

Reply:
UPDATE: @spbrogan I could not build -NOOPT with your suggestion here. I needed to provide the relative path. The host based unit test could not see the protocol library since I was just creating a small executable with NVMe and unit test source.

Reply:
that doesn't make sense to me.  I'll look into this one.

Reply:
I was able to build and pass the tests with this change
``` c
//sean#include "../../../../Include/Protocol/MediaSanitize.h"
#include <Protocol/MediaSanitize.h>
```

Reply:
Thanks for looking into this one @spbrogan. I figured out what the issue was for me. 

**When I added the #include line before the other relative NVM Express include paths** (like below), I got the compile error.

#include <Uefi.h>
#include <Library/BaseLib.h>
#include <Library/BaseMemoryLib.h>
#include <Library/UefiLib.h>
#include <Library/DebugLib.h>
#include <Library/UnitTestLib.h>
#include <Library/UefiBootServicesTableLib.h>
#include <Library/UefiApplicationEntryPoint.h>
#include <Library/UefiRuntimeServicesTableLib.h>
#include <Library/UefiBootManagerLib.h>
#include <Protocol/NvmExpressPassthru.h>
**#include <Protocol/MediaSanitize.h>**

#include "../NvmExpress.h"
#include "../NvmExpressBlockIo.h"
#include "../NvmExpressMediaSanitize.h"
#include "../NvmExpressHci.h"

However, when I add the MediaSanitize.h header include **after the local, relative NVM Express header includes** (see below), I got a successful compile and run.

I will update the file to use the public include. Thanks!

Reply:
after more review you have fixed it correctly by finding the real dependency (protocol/blockio.h).  Might be helpful to those trying to use the protocol if a comment was added in the header file saying the definition depends on blockio.h.  

if you have no other changes to make...please disregard and just resolve.  :)